policy-controller/k8s/index/src/http_route.rs

```diff
@@ -1,20 +1,74 @@
 use anyhow::{bail, Error, Result};
-use k8s_gateway_api::ParentReference;
-use linkerd_policy_controller_core::http_route::{
-    HeaderMatch, Hostname, HttpFilter, HttpRoute, HttpRouteMatch, HttpRouteRule, PathMatch,
-    PathModifier, QueryParamMatch, Value,
-};
+use k8s_gateway_api as api;
+use linkerd_policy_controller_core::http_route;
 
 #[derive(Clone, Debug, PartialEq)]
-pub struct RouteBinding {
-    pub route: HttpRoute,
-    pub parent_refs: Vec<ParentReference>,
+pub struct InboundRouteBinding {
+    pub parents: Vec<InboundParentRef>,
+    pub route: http_route::InboundHttpRoute,
 }
 
-impl TryFrom<k8s_gateway_api::HttpRoute> for RouteBinding {
+#[derive(Clone, Debug, PartialEq)]
+pub enum InboundParentRef {
+    Server(String),
+}
+
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum InvalidParentRef {
+    #[error("HTTPRoute resource does not reference a Server resource")]
+    DoesNotSelectServer,
+
+    #[error("HTTPRoute resource may not reference a parent Server in an other namespace")]
+    ServerInAnotherNamespace,
+
+    #[error("HTTPRoute resource may not reference a parent by port")]
+    SpecifiesPort,
+
+    #[error("HTTPRoute resource may not reference a parent by section name")]
+    SpecifiesSection,
+}
+
+impl TryFrom<api::HttpRoute> for InboundRouteBinding {
     type Error = Error;
 
-    fn try_from(route: k8s_gateway_api::HttpRoute) -> Result<Self, Self::Error> {
+    fn try_from(route: api::HttpRoute) -> Result<Self, Self::Error> {
+        let parents = route
+            .spec
+            .inner
+            .parent_refs
+            .into_iter()
+            .flatten()
+            .map(
+                |api::ParentReference {
+                     group,
+                     kind,
+                     namespace,
+                     name,
+                     section_name,
+                     port,
+                 }| {
+                    if port.is_some() {
+                        return Err(InvalidParentRef::SpecifiesPort);
+                    }
+                    if section_name.is_some() {
+                        return Err(InvalidParentRef::SpecifiesSection);
+                    }
+
+                    if group.as_deref() != Some("policy.linkerd.io")
```

Comment:
Just to confirm my understanding, this is one place where the policy.linkerd.io HttpRoute diverges from the gateway API HttpRoute?  That is, the gateway API HttpRoute could have parent refs which target various kinds (e.g. Gateways etc.) but the policy.linkerd.io may only target resources that we explicitly support?

Reply:
That's right. But in the current form this would only prevent us from _indexing_ HTTP routes that target types. I.e. we wouldn't actually prevent them from being created, we just won't keep them cached... 

As I think about this -- it's probably better to just skip any parent ref we don't know about and fail the index if we don't have any remaining parent refs.